manifests/repo/openshift-server.pp

```diff
@@ -5,8 +5,8 @@
 # when setting 'install_method' to 'none' in addition with 'yum::repo::epel' and 'yum::repo::jenkins' 
 #
 class yum::repo::openshift-server ($version=4) {
-  if $lsbdistrelease !~ /^6/ {
-    warning('This module works only for RHEL6')
+  if $operatingsystemrelease !~ /^6/ {
```

Comment:
Always better to refer to facts with the fully qualified name: $::operatingsystemrelease